lib/practice_exercises.rb

```diff
@@ -1,13 +1,49 @@
+require "pry"
+# Time Complexity: o(n)
+# Space Complexity: o(1)
 
-# Time Complexity: ?
-# Space Complexity: ?
+# num = [1, 2, 2, 3, 4, 4]
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+  i = 0
+  while i <= list.length
+    if list[i] == list[i + 1]
+      list.delete_at(i)
+    end
+    i += 1
+  end
+
+  # print list.length
+  return list
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# remove_duplicates(num)
+#==============================================================
+
+# Time Complexity: o(n^2)
+# Space Complexity: ?o(n)
+
 def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
-end
+  total_words = strings.length
+  prefix = ""
 
+  if total_words == 0
+    return prefix
+  end
+
+  if total_words == 1
+    return strings[0]
+  end
+
+  total_words.times do |index|
```

Comment:
shouldn't one of these loops go the number of characters in a string, rather than the number of strings times?

You also have an issue with the `if` statement, you're comparing the wrong things.